ionic-team/stencil
# feat(typescript): Update to 5.5.4

## What is the current behavior?

Since stencil `v4.20.0`, if you set `verbatimModuleSyntax: true` in your tsconfig, you get unused errors for `h` and `Fragment` in every tsx file:

```
[ ERROR ]  TypeScript: file/path.tsx:21:1
           'h' is declared but its value is never read.

      L1:  import { h, type FunctionalComponent } from '@stencil/core';
```

This is caused by a bug in typescript, where `jsxFactory` and `jsxFragmentFactory` were not being marked as used: https://github.com/microsoft/TypeScript/issues/59117

This bug has been fixed and released in `v5.5.4`.


## What is the new behavior?

Updating stencil to use typescript `v.5.5.4` allows you to use `verbatimModuleSyntax`.


## Documentation

_none_

## Does this introduce a breaking change?


## Testing

- Create a fresh Stencil project `yarn create stencil` 
- Add  `"verbatimModuleSyntax": true` to the `tsconfig.json`
- Optionally 
   - `"jsxFragmentFactory": "Fragment"` to the `tsconfig.json`
   - Add fragment usage to `my-component.tsx`
- Bulid and observe error:
![image](https://github.com/user-attachments/assets/66c41116-fff6-4c6d-ad50-0f7786f2010a)
- Update stencil in package.json to use changes in this PR.
- Build and observe success:
![image](https://github.com/user-attachments/assets/eebb49ca-b1d1-450d-ba96-594aba8287bb)



## Other information
[Discord discussion](https://discord.com/channels/520266681499779082/1278638564074192936)
